app/src/main/java/com/b_lam/resplash/ui/photo/zoom/PhotoZoomActivity.kt

```diff
@@ -19,34 +20,54 @@ class PhotoZoomActivity : BaseActivity(R.layout.activity_photo_zoom) {
 
     override val binding: ActivityPhotoZoomBinding by viewBinding()
 
+    private var isSystemUiVisible = false
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+        toggleSystemUI(false)
+
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-            window.setDecorFitsSystemWindows(false)
-            window.insetsController?.let {
-                it.hide(WindowInsets.Type.statusBars() or WindowInsets.Type.navigationBars())
-                it.systemBarsBehavior = WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+            window.decorView.setOnApplyWindowInsetsListener { view, windowInsets ->
+                val suppliedInsets = view.onApplyWindowInsets(windowInsets)
+                isSystemUiVisible = suppliedInsets.isVisible(
+                    WindowInsetsCompat.Type.statusBars()
+                            or WindowInsetsCompat.Type.navigationBars()
+                )
+                suppliedInsets
             }
         } else {
-            window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-                    or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                    or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                    or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-                    or View.SYSTEM_UI_FLAG_FULLSCREEN
-                    or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
+            window.decorView.setOnSystemUiVisibilityChangeListener {
+                isSystemUiVisible = (it and View.SYSTEM_UI_FLAG_HIDE_NAVIGATION) == 0
+            }
+        }
+
+        binding.zoomImageView.setOnClickListener {
+            toggleSystemUI(!isSystemUiVisible)
         }
 
         val url = intent.getStringExtra(EXTRA_PHOTO_URL)
 
         url?.let {
             binding.zoomImageView.loadPhotoUrl(it)
-        }  ?: run {
+        } ?: run {
             toast(R.string.oops)
             finish()
         }
     }
 
+    private fun toggleSystemUI(showSystemUI: Boolean) {
```

Comment:
small nit: `toggleSystemUI` and `showSystemUI` should be `toggleSystemUi` and `showSystemUi` to be consistent with `isSystemUiVisible` above